src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -154,7 +156,14 @@ protected set
             }
         }
 
-        internal abstract GreenNode GetSlot(int index);
+        internal abstract GreenNode? GetSlot(int index);
+
+        internal GreenNode GetRequiredSlot(int index)
+        {
+            var node = GetSlot(index);
+            RoslynDebug.Assert(node is object);
```

Comment:
💡 I noticed this helper recently. Doesn't map back to `Debug` but still usable.

```suggestion
            RoslynDebug.AssertNotNull(node);
```

Reply:
The not mapping back to `Debug` is why I prefer the current form. 

---
In reply to: [363546862](https://github.com/dotnet/roslyn/pull/40767#discussion_r363546862) [](ancestors = 363546862)